adapters/src/coinmarketcap.ts

```diff
@@ -0,0 +1,21 @@
+import { AssetPrice, newAssetPrice, fetchAsJson, FetchResponse } from '@defichain/salmon-fetch'
+import BigNumber from 'bignumber.js'
+const COINMARKETCAP_URL = 'https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/latest'
+
+export default async function (symbols: string[], apiToken: string): Promise<AssetPrice[]> {
+  const fetchPath = `${COINMARKETCAP_URL}?symbol=${symbols.join(',')}&CMC_PRO_API_KEY=${apiToken}`
+  const res: FetchResponse = await fetchAsJson(fetchPath, {
+    method: 'GET'
+  })
+  const json = res.data.data
+  return Object.keys(json).map((asset: any) => {
+    const data = json[asset]
+    const timestamp = new BigNumber(Date.parse(data.last_updated)).multipliedBy(1000)
```

Comment:
Why are we multiplying by 1000 here? I think that's a bug, it's not in the original piece

Reply:
understand.. just div(1000) for blockchain ts in seconds format (10 digits)